src/components/approval-type/ApprovalType.vue

```diff
@@ -117,148 +114,147 @@ export default class ApprovalType extends Vue {
-  private valid = false
-  private approvalTypeSelected = ''
   private noticeDateText = ''
   private applicationDateText = ''
 
   // For template
   readonly ApprovalTypes = ApprovalTypes
 
-  // Date Picker Rules
-  protected readonly datePickerRules = [(v: string) => !!v || 'Date is required']
+  /** Date picker rules. */
+  readonly datePickerRules = [(v: string) => !!v || 'Date is required']
 
-  // Text Field Rules
-  protected readonly courtOrderNumRules = [
+  /** Text field rules. */
+  readonly courtOrderNumRules = [
     (v: string) => (!v || !/^\s/g.test(v)) || 'Invalid spaces', // leading spaces
     (v: string) => (!v || !/\s$/g.test(v)) || 'Invalid spaces', // trailing spaces
-    (v: string) => (!v || !(v.length < 5)) || 'Court order number is invalid',
-    (v: string) => (!v || !(v.length > 20)) || 'Court order number is invalid',
+    (v: string) => (!v || !(v.length < 5)) || 'Court order number is too short',
+    (v: string) => (!v || !(v.length > 20)) || 'Court order number is too long',
     (v: string) => !!v || 'A Court Order number is required'
   ]
 
   /** Called when component is mounted. */
   mounted (): void {
-    // Copy props to mutable properties
+    // copy props to mutable properties
     if (this.approvedByRegistrar) {
-      this.courtOrderNumberText = ''
-      this.approvalTypeSelected = ApprovalTypes.VIA_REGISTRAR
-    } else if (this.courtOrderNumber) {
-      this.courtOrderNumberText = this.courtOrderNumber
-      this.approvalTypeSelected = ApprovalTypes.VIA_COURT_ORDER
-    } else {
-      // Default state (no button selected)
-      this.radioButtonChanged('')
+      // first set approval type, then set values
+      // otherwise, validation will fail
+      this.setApprovalTypeSelected(ApprovalTypes.VIA_REGISTRAR)
+      this.setNoticeDateText(this.noticeDate)
+      this.setApplicationDateText(this.applicationDate)
+    } else if (this.approvedByCourtOrder) {
+      // first set approval type, then set values
+      // otherwise, validation will fail
+      this.setApprovalTypeSelected(ApprovalTypes.VIA_COURT_ORDER)
+      this.setCourtOrderNumberText(this.courtOrderNumber)
     }
-    if (!this.isCourtOrderRadio) {
-      this.approvalTypeSelected = ApprovalTypes.VIA_COURT_ORDER
+  }
+
+  /** When validate flag is set, validates the component. */
+  @Watch('validate')
+  private async onValidateChanged (validate: boolean): Promise<void> {
+    if (validate) {
+      // wait for rules to get processed before validating
+      await Vue.nextTick()
+      this.validateComponent()
     }
   }
 
-  /** Triggers the form validation. */
-  public validate (): boolean {
+  /** Validates the component and emits the validity. */
+  @Emit('valid')
+  private validateComponent (): boolean {
+    let valid = true
     if (this.approvalTypeSelected === ApprovalTypes.VIA_COURT_ORDER) {
-      let status = this.$refs.courtNumRef.validate()
-      this.$emit('valid', status)
```

Comment:
We shouldn't use this all over the code. We should have a single place where a property is emitted. This is now THIS method (as per `@Emit` decorator).